### PR TITLE
Don't show section code for provider-managed sections

### PR DESCRIPTION
**Pull request by joshlory**
This PR depends on PR https://github.com/code-dot-org/code-dot-org/pull/16556.

screen shot 2017-07-21 at 4 20 05 pm

+cc @poorvasingal

### CODE REVIEW

**Comment by joshlory** [comment]
There are still a few places we show the section code:
 - The section preview table on the home page.  That table is getting replaced by this one — should I still update it?
 - The AngularJS section manage page.  Same question (but maybe a different answer).

**Comment by Bjvanminnen** [comment]
The section preview table on the home page. That table is getting replaced by this one — should I still update it?
- Depends how soon it's being replaced possibly. Might also be a question for PM.

The AngularJS section manage page. Same question (but maybe a different answer).
- Again, might be a question for PM. What do Google/Clever sections see if we dont make any changes?

File: apps/src/templates/teacherDashboard/SectionRow.jsx [hunk 1]
```diff
@@ -245,6 +245,15 @@ class SectionRow extends Component {
...
     const persistedSection = !!section.code;
 
+    let sectionCode = '';
+    if (!editing) {
+      if (section.providerManaged) {
+        sectionCode = '---';
+      } else {
+        sectionCode = section.code;
+      }
+    }
```
Comment (Bjvanminnen) [inline apps/src/templates/teacherDashboard/SectionRow.jsx]: Should we just have the server return '---' for section.code when provider managed? or do we need to use the code elsewhere?

Comment (joshlory) [reply apps/src/templates/teacherDashboard/SectionRow.jsx]: I'd like to update the '---' with a little `fa-question-circle` icon and a tooltip explaining why the section doesn't have a code.  Maybe not needed...

**Comment by joshlory** [comment]
In both cases they'll still see the `G-12345` code (but it won't be joinable because of the work in PR https://github.com/code-dot-org/code-dot-org/pull/16556).

So not a terrible user experience, but worth fixing if we're not going to deprecate those views soon.  I'll plan to do that in a separate PR!

**Comment by joshlory** [comment]
Sample tooltip:

screen shot 2017-07-24 at 11 37 17 am

File: apps/src/templates/teacherDashboard/SectionRow.jsx (hunk 1 above)
Comment (joshlory) [reply apps/src/templates/teacherDashboard/SectionRow.jsx]: Updated in https://github.com/code-dot-org/code-dot-org/pull/16557/commits/5be4ec75e916f2427002c130ac329c132e774270.

**Comment by Bjvanminnen** [comment]
Wonder if that question mark icon should be bigger (but also maybe a lighter gray?) so that it's easier to mouse over

**Comment by joshlory** [comment]
Good call, I changed the icon and made the hover target the entire element:

screen shot 2017-07-24 at 12 06 23 pm

**Pull request merged**